chain/network/src/routing/routing_table_actor.rs

```diff
@@ -306,63 +306,56 @@ impl RoutingTableActor {
-                } else {
-                    None
-                }
+        // TODO - use drain_filter once it becomes stable - #59618
+        let peers_to_remove = (self.peer_last_time_reachable.iter())
+            .filter(|(_, &last_time)| {
+                now.saturating_duration_since(last_time) >= prune_edges_not_reachable_for
             })
+            .map(|(peer_id, _)| peer_id.clone())
             .collect::<HashSet<_>>();
 
-        // Save nodes on disk and remove from memory only if elapsed time from oldest peer
-        // is greater than `SAVE_PEERS_MAX_TIME`
-        if !force_pruning && now.saturating_duration_since(oldest_time) < SAVE_PEERS_MAX_TIME {
-            return Vec::new();
-        }
         debug!(target: "network", "try_save_edges: We are going to remove {} peers", peers_to_remove.len());
 
-        let current_component_nonce = self.next_available_component_nonce;
-        self.next_available_component_nonce += 1;
-
         let mut update = self.store.store_update();
-        // Stores next available nonce.
-        let _ = update.set_ser(ColLastComponentNonce, &[], &self.next_available_component_nonce);
 
         // Sets mapping from `peer_id` to `component nonce` in DB. This is later used to find
         // component that the edge belonged to.
         for peer_id in peers_to_remove.iter() {
             let _ = update.set_ser(
                 ColPeerComponent,
                 peer_id.try_to_vec().unwrap().as_ref(),
-                &current_component_nonce,
+                &self.next_available_component_nonce,
             );
 
             self.peer_last_time_reachable.remove(peer_id);
         }
 
-        let component_nonce = index_to_bytes(current_component_nonce);
-        let edges_to_remove = self
-            .edges_info
-            .iter()
-            .filter_map(|(key, edge)| {
-                if peers_to_remove.contains(&key.0) || peers_to_remove.contains(&key.1) {
-                    Some(edge.clone())
-                } else {
-                    None
-                }
-            })
+        // TODO - use drain_filter once it becomes stable - #59618
+        let edges_to_remove = (self.edges_info.iter())
+            .filter(|(key, _)| peers_to_remove.contains(&key.0) || peers_to_remove.contains(&key.1))
+            .map(|(_, edge)| edge.clone())
             .collect();
 
-        let _ = update.set_ser(ColComponentEdges, component_nonce.as_ref(), &edges_to_remove);
+        let _ = update.set_ser(
+            ColComponentEdges,
+            index_to_bytes(self.next_available_component_nonce).as_ref(),
+            &edges_to_remove,
+        );
+
+        self.next_available_component_nonce += 1;
+        // Stores next available nonce.
+        let _ = update.set_ser(ColLastComponentNonce, &[], &self.next_available_component_nonce);
 
         if let Err(e) = update.commit() {
             warn!(target: "network", "Error storing network component to store. {:?}", e);
```

Comment:
unrelated to PR, but feels a bit odd that we we update in-memory hash maps even if ommid didn't succeed.

Reply:
@matklad That's a good point. However, there isn't much we can do when a database commit fails, we don't handle database failures anywhere in our code.